python_modules/libraries/dagster-dlt/dagster_dlt/components/dlt_load_collection/component.py

```diff
@@ -46,17 +87,14 @@ class DltLoadSpecModel(Resolvable):
             model_field_type=str,
         ),
     ]
-
-
-class ComponentDagsterDltTranslator(DagsterDltTranslator):
-    """Custom base translator, which generates keys from dataset and table names."""
-
-    def get_asset_spec(self, data: DltResourceTranslatorData) -> AssetSpec:
-        table_name = data.resource.table_name
-        if isinstance(table_name, Callable):
-            table_name = data.resource.name
-        prefix = [data.pipeline.dataset_name] if data.pipeline else []
-        return super().get_asset_spec(data).replace_attributes(key=AssetKey(prefix + [table_name]))
+    translator: Annotated[
+        Optional[DagsterDltTranslator],
+        Resolver(
+            resolve_translator,
+            model_field_name="asset_attributes",
+            model_field_type=Union[str, AssetAttributesModel],
+        ),
+    ] = None
```

Comment:
did you see https://github.com/dagster-io/dagster/pull/29171 ? in order to provide a nicer pythonic interface, we opted towards a `translation` function `(AssetSpec, DomainInfo) -> AssetSpec` and backgrounded the translator class

Reply:
ah neat, updated to match